modules/core/src/main/java/org/apache/synapse/config/xml/EnrichMediatorFactory.java

```diff
@@ -187,4 +200,49 @@ public QName getTagQName() {
         return XML_Q;
     }
 
+    private void validateTypeCombination(OMElement sourceEle, OMElement targetEle) {
+        int sourceType = -1;
+        int targetType = -1;
+
+        // source type attribute
+        OMAttribute sourceTypeAttr = sourceEle.getAttribute(ATT_TYPE);
+        if (sourceTypeAttr != null && sourceTypeAttr.getAttributeValue() != null) {
+            sourceType = convertTypeToInt(sourceTypeAttr.getAttributeValue());
+
+            // Source type is different form the existing (0-custom, 1-envelope, 2-body, 3-property, 4-inline)
+            if (sourceType < 0) {
+                logger.info("Un-expected source type");
+            }
+        }
+
+        // target type attribute
+        OMAttribute targetTypeAttr = targetEle.getAttribute(ATT_TYPE);
+        if (targetTypeAttr != null && targetTypeAttr.getAttributeValue() != null) {
```

Comment:
can targetTypeAttr being null a valid use case ?, if not we need to throw an exception

Reply:
Done